### ISSUE TITLE
After delays do not work on broadcast connections in the C target

### ISSUE THREAD

**Issue by cmnrd** [lf-lang/lingua-franca#554]
This is closely related to #553. While the fix implemented in #533 was sufficient to solve the problem for C++, it persists in C (and probably Python). I have implemented a [BroadcastAfter.lf](https://github.com/lf-lang/lingua-franca/blob/broadcast-after-c/test/C/src/multiport/BroadcastAfter.lf) test and pushed it to the `broadcast-after-c` branch (which is based on #533) to illustrate the issue. Oddly, I see the following warning when compiling the test in lfc:
```
lfc: warning: Destination is wider than the source. Inputs will be missing.
 --> test/C/src/multiport/BroadcastAfter.lf:41:5
   |
40 |     b = new[4] Destination();
41 |     (a.out)+ -> b.in after 1 sec;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Destination is wider than the source. Inputs will be missing.
   |
42 | }
```
This warning does not appear for the C++ version (neither on master nor with #553).

**Comment by edwardalee** [comment]
In Epoch, this same file throws an NPE rather than reporting the same error. The NPE is in EclipseErrorReporter.reportError on [this line](https://github.com/lf-lang/lingua-franca/blob/c4126527405b4d71c17d19e28ce4a84d7ca36e08/org.lflang/src/org/lflang/generator/EclipseErrorReporter.java#L70). I tried to fix this, but oddly, just refactoring this line into two parts doesn't work:
```
        final ICompositeNode node = NodeModelUtils.getNode(obj);
        final int line = node.getStartLine();
```
The file no longer compiles with a very odd error message:
```
The type org.eclipse.core.runtime.jobs.ISchedulingRule cannot be resolved. It is indirectly referenced from required .class 
```
And the compile error is on a completely unrelated line in the file.

Also, since we have so many branches going simultaneously, I have no idea where to try to fix this or whether it has been fixed already.  But in any case, I really don't think we should be importing any xtext classes (`NodeModelUtils`).
Reporting errors needs to be robust.  Help?

**Comment by cmnrd** [comment]
I think a good place to fix the bug described in this issue is the `broadcast-after-c` that I already started to have a place for the broken C test.

I think that the error reporting issue that you report should be solved separately. It is likely that this error occurs now because of the version increment of our dependencies in #542. I am actually surprised that this did not fail before, as clearly `obj` can be `null`, but there is no null check in place.

As to your compile error: Maybe the problem is that the `broadcast-after-c` is based on a relatively new master, which contains the version upgrades of #542. So maybe your installation does not fit the dependencies listed in the build files anymore. Or maybe your Eclipse got confused from switching between branches that require different library versions.

**Comment by cmnrd** [comment]
I fixed the NPE in #559. Epoch now shows the error "null - See error log for details" for the C program. I am not sure what this means or where it comes from.

**Comment by cmnrd** [comment]
I now fixed the NPE problem for real in #554. This also fixes the compile error you mention.

In afterthought, I think that the solution I implemented in #553 isn't perfect. For a connection of a single port on the left to N ports on the right, the solution from #553 will create N delay reactors. It would be more efficient to create a single delay reactor and then broadcast the delay output.

**Comment by edwardalee** [comment]
I still get the NPE.

**Comment by cmnrd** [comment]
I am sorry, I inserted the wrong link above. The NPE fix is implemented in #562. Did you try with this branch?

**Comment by lhstrh** [comment]
Just merged the NPE fix into master.
